src/test/java/pl/byczazagroda/trackexpensesappbackend/controller/WalletControllerTest.java

```diff
@@ -89,6 +200,45 @@ void shouldThrowAnExceptionWhenNameIsTooLong() throws Exception {
         result.andExpect(status().isBadRequest());
     }
 
+    @Test
+    void itShouldReturnBadRequestWhenIdIsZero() throws Exception {
+        //GIVEN
+        Instant timeCreated = Instant.now();
+
+        UpdateWalletDTO updateWalletDTO = new UpdateWalletDTO(0L, "@#$%^&");
+        given(walletService.updateWallet(updateWalletDTO))
+                .willReturn(new WalletDTO(0L, "", timeCreated));
+
+        //WHEN
+        ResultActions result = mockMvc.perform(
+                put("/api/wallet")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(Objects.requireNonNull(objectMapper.writeValueAsString(updateWalletDTO)))
+        );
+
+        //THEN
+        result.andExpect(status().isBadRequest());
+    }
+
+    @Test
+    void itShouldReturnBadRequestWhenIdIsNegative() throws Exception {
+        //GIVEN
+        Instant timeCreated = Instant.now();
+
+        UpdateWalletDTO updateWalletDTO = new UpdateWalletDTO(-1L, "@#$%^&");
+        given(walletService.updateWallet(updateWalletDTO))
+                .willReturn(new WalletDTO(-1L, "", timeCreated));
+
+        //WHEN
+        ResultActions result = mockMvc.perform(
+                put("/api/wallet")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(Objects.requireNonNull(objectMapper.writeValueAsString(updateWalletDTO)))
+        );
+
```

Comment:
Why you didn't describe `//then` section?

Reply:
Probably disappeared during merging with develop branch. '//then' section added.